src/monodroid/jni/monodroid-glue.cc

```diff
@@ -652,6 +652,7 @@ MonodroidRuntime::mono_runtime_init ([[maybe_unused]] dynamic_local_string<PROPE
 		log_error (LOG_DEFAULT, "Failed to parse runtime args: '%s'", runtime_args.get ());
 	} else if (options.debug && cur_time > options.timeout_time) {
 		log_warn (LOG_DEBUGGER, "Not starting the debugger as the timeout value has been reached; current-time: %lli  timeout: %lli", cur_time, options.timeout_time);
+		delete[] options.host;
```

Comment:
This should happen before the `#else` on line 737.

Additionally, the declaration of `options` on line 646 should become:

```cpp
RuntimeOptions options{};
```